cmd/inspektor-gadget/cmd/health.go

```diff
@@ -77,57 +84,67 @@ func execPodQuick(client *kubernetes.Clientset, node string, podCmd string) stri
 	}
 	podName := pods.Items[0].Name
 
-	kubectlCmd := fmt.Sprintf("kubectl ")
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	loadingRules.DefaultClientConfig = &clientcmd.DefaultClientConfig
 	if viper.GetString("kubeconfig") != "" {
-		kubectlCmd += "--kubeconfig=" + viper.GetString("kubeconfig")
+		loadingRules.ExplicitPath = viper.GetString("kubeconfig")
 	}
-	kubectlCmd += fmt.Sprintf(" exec -n kube-system %s -- %s", podName, podCmd)
+	overrides := &clientcmd.ConfigOverrides{ClusterDefaults: clientcmd.ClusterDefaults}
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, overrides)
 
-	cmd := exec.Command("/bin/sh", "-c", kubectlCmd)
-	stdoutStderr, err := cmd.CombinedOutput()
-	if err != nil {
-		if len(stdoutStderr) != 0 {
-			return fmt.Sprintf("%s\n%s", err, stdoutStderr)
-		} else {
-			return fmt.Sprintf("%s", err)
-		}
-	}
-
-	return fmt.Sprintf("%s", string(stdoutStderr))
-}
-
-func execPodQuickStart(client *kubernetes.Clientset, node string, podCmd string) string {
-	var listOptions = metaV1.ListOptions{
-		LabelSelector: "k8s-app=gadget",
-		FieldSelector: "spec.nodeName=" + node + ",status.phase=Running",
-	}
-	pods, err := client.CoreV1().Pods("kube-system").List(listOptions)
+	restConfig, err := clientConfig.ClientConfig()
 	if err != nil {
 		return fmt.Sprintf("%s", err)
 	}
-	if len(pods.Items) == 0 {
-		return "not-found"
-	}
-	if len(pods.Items) != 1 {
-		return "too-many"
-	}
-	podName := pods.Items[0].Name
-
-	kubectlCmd := fmt.Sprintf("kubectl ")
-	if viper.GetString("kubeconfig") != "" {
-		kubectlCmd += "--kubeconfig=" + viper.GetString("kubeconfig")
-	}
-	kubectlCmd += fmt.Sprintf(" exec -n kube-system %s -- %s", podName, podCmd)
-
-	cmd := exec.Command("/bin/sh", "-c", kubectlCmd)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
-	err = cmd.Start()
+	factory.SetKubernetesDefaults(restConfig)
+	restClient, err := restclient.RESTClientFor(restConfig)
+	if err != nil {
+		return fmt.Sprintf("RESTClientFor: %s", err)
+	}
+	req := restClient.Post().
+		Resource("pods").
+		Name(podName).
+		Namespace("kube-system").
+		SubResource("exec").
+		Param("container", "gadget").
+		VersionedParams(&corev1.PodExecOptions{
+			Container: "gadget",
+			Command:   []string{"/bin/sh", "-c", podCmd},
+			Stdin:     false,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       false,
+		}, scheme.ParameterCodec)
+
+        exec, err := remotecommand.NewSPDYExecutor(restConfig, "POST", req.URL())
 	if err != nil {
 		return fmt.Sprintf("%s", err)
 	}
 
-	return ""
+	if capture {
+		var stdout, stderr bytes.Buffer
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  nil,
+			Stdout: &stdout,
+			Stderr: &stderr,
+			Tty:    false,
+		})
+		if err != nil {
+			return fmt.Sprintf("%s", err) + stdout.String() + stderr.String()
+		}
+		return stdout.String() + stderr.String()
```

Comment:
I think the prototype of the function `execPod()` could be improved by returning a Golang `error` as well. So we would propagate the error in `install.go` and the command `inspektor-gadget install` could return a proper exit value.

Reply:
Ok, I've prepared everything for stdout postprocessing through a custom `io.Writer` that, e.g., can read the first lines to detect failures and then prepend node identifiers (But that should be in another commit/PR).